### ISSUE TITLE
BTP supporting multiple identity providers

### ISSUE THREAD

**Issue by sergePSE** [ui5-community/wdi5#510]
**Describe the bug**
During navigation to the baseUrl there might be called BCP authentication page with multiple options.

Only after selecting some option BTP authentication can be continued

**To Reproduce**

1. Go to BTP restricted page with several identity providers

**Expected behavior**
Selecting default identify provider if not specified or allow identity provider selection in addition to the user/pass. Options are described as plain html <a/> tags like 
```
<a href="saml/discovery?returnIDParam=idp&amp;entityID=https://space-tenant.authentication.sap.hana.ondemand.com&amp;idp=accounts.ondemand.com&amp;isPassive=true" class="saml-login-link">accounts01.ondemand.com</a>
```
**Screenshots**
![Screenshot 2023-08-01 at 16 42 52](https://github.com/ui5-community/wdi5/assets/18466995/07001da1-4cae-4689-b9c4-f9668e6543f4)

**Runtime Env (please complete the following information):**

- `wdi5/wdio-ui5-service`-version: 1.5.1
- `wdio`-version (output of `wdio --version`): 7.3.2
- `node`-version (output of `node --version`): 18.16.1
- OS: Mac
- Browser + Version chrome 115

**Additional context**
Add any other context about the problem here,
e.g. any options the target browser is started with like `--headless` or
if the tests run in a CI environment

**Comment by dominikfeininger** [comment]
yeah, would be a really nice feature to have 👍. Currently the wdi5 code team is quite busy. More than welcome to add this feature yourself. See the [contribution guid](https://ui5-community.github.io/wdi5/#/contributing) in the docs.

**Comment by huahouye** [comment]
ust a workaround (not a solution) here in case anyone needs this.

```sh
npm install https://github.com/weikaopu/wdi5.git#v2.1.1-feat-multiple-idp
```

<img width="1358" alt="Image" src="https://github.com/user-attachments/assets/3468e70b-3579-48ec-b348-19c0645e8a5f" />

<img width="569" alt="Image" src="https://github.com/user-attachments/assets/7df97e06-302e-4f99-b0ac-65e49590b3cb" />

**Comment by vobu** [comment]
thanks for the hints - does that custom wdi5 version in https://github.com/weikaopu/wdi5 contain any specific code for choosing an IdP?

**Comment by huahouye** [comment]
Not need any specific code for choosing an IdP at all, but need to config the domain-only part of the IAS tenant URL for `wdi5.capabilities."wdi5:authentication".idpDomain` in my `wdi5.conf.ts`. 

Actually it is quite easy to implement this workaround. I just need to find the \<a\> tag that contains `idp=my-idpDomain` and click on it if it is there before entering username/apssword in [BTPAuthenticator.ts#L38](https://github.com/weikaopu/wdi5/blob/883b3efd7c8741cf2543ab59bc4bc00ad61a889b/src/lib/authentication/BTPAuthenticator.ts#L38)

**Comment by vobu** [comment]
> Not need any specific code for choosing an IdP at all, but need to config the domain-only part of the IAS tenant URL for `wdi5.capabilities."wdi5:authentication".idpDomain` in my `wdi5.conf.ts`.
> 
> Actually it is quite easy to implement this workaround. I just need to find the <a> tag that contains `idp=my-idpDomain` and click on it if it is there before entering username/apssword in [BTPAuthenticator.ts#L38](https://github.com/weikaopu/wdi5/blob/883b3efd7c8741cf2543ab59bc4bc00ad61a889b/src/lib/authentication/BTPAuthenticator.ts#L38)

waaaah, this is so 🆒 - why don't you throw this into a PR and we'll take care of providing a test setup for that?! 
Pretty please 🥺 😸

**Comment by vobu** [comment]
closed by #673 and #674
